### ISSUE TITLE
SLIM for tetmesh improvement

### ISSUE THREAD

**Issue by mlivesu** [libigl/libigl#2401]
#### Describe the bug

Hi, I am trying to use SLIM for tetmesh improvement (as shown in Fig.18 in the original paper). Apparently `igl::SLIMData` contains a bool flag `mesh_improvement_3d` that allows to set the canonical tetrahedron as a reference element for the distortion optimization, but it is impossible to set the flag correctly from the outside because `igl::slim_precompute` overwrites its value (line 785) and then uses it to compute the deformation gradients (line 790), using the input tets as a reference shape regardless of your intent.

Setting `mesh_improvement_3d = true` from the outside and commenting line 785 I was hoping to obtain the wanted result, but I still see that something is off. Attached is the interior of a simple tetmesh before/after the optimization (I am holding the boundary vertices with weight 10). I have no clue of why only the outer layer of tetrahedra is improving, whereas the interior ones do not improve their aspect ratio. Perhaps I am doing something wrong? 

<img width="1482" alt="before_after" src="https://github.com/libigl/libigl/assets/7881910/fac793dd-c359-497d-9862-4469dbcbfb2f">

The code I am using looks like (I can share the whole project if necessary): 

```c++
igl::SLIMData data;
Eigen::MatrixXd V;
Eigen::MatrixXi F;
Eigen::VectorXi b; 
Eigen::MatrixXd bc;
// fill V, F, b, bc with vertices, tets, indices of srf vertices and srf coordinates, respectively
data.mesh_improvement_3d = true;
slim_precompute(V, F, V, data, igl::MappingEnergyType::SYMMETRIC_DIRICHLET, b, bc, 10);
slim_solve(data,1);
```

The mesh is [A.mesh.zip](https://github.com/user-attachments/files/16143362/A.mesh.zip)

#### Platform

- [ ] Windows
- [x] macOS
- [ ] Linux

**Comment by alecjacobson** [comment]
It's unlikely that I'll personally make time to look into this. @danielepanozzo or @schuellc might know what's going on in that file.

**Comment by alecjacobson** [comment]
Oh, I see it's actually @MichaelRabinovich .

**Comment by MichaelRabinovich** [comment]
It's been quite a while. Have you tried lowering the boundary vertices weight to see if this change?

**Comment by mlivesu** [comment]
Hi Alec, I opened the issue mostly because the overwriting of the input parameter is definitely a bug. For the weird behavior of the solver I was basically taking my chances, in case somebody experienced similar issues in the past :)

**Comment by mlivesu** [comment]
> It's been quite a while. Have you tried lowering the boundary vertices weight to see if this change?

yep. With no boundary conditions also the surface evolves, improving the shape of the outer tetrahedra. Still, the inner tests stay as they are. Since this is essentially local/global optimization I was wondering whether it may be due to the fact that way too many local solutions are trying to move to poor high valence vertex in the middle of the sphere, which "locks". But this is just me guessing...
